klippy/klippy.py

```diff
@@ -131,6 +142,17 @@ def _read_config(self):
             m.add_printer_objects(config)
         # Validate that there are no undefined parameters in the config file
         pconfig.check_unused_options(config)
+    def _handle_info_request(self, web_request):
+        version = self.start_args['software_version']
+        cpu_info = self.start_args['cpu_info']
+        error = self.state_message != message_startup and \
+            self.state_message != message_ready
+        web_request.send(
+            {'cpu': cpu_info, 'version': version,
+                'hostname': socket.gethostname(),
+                'is_ready': self.state_message == message_ready,
+                'error_detected': error,
+                'message': self.state_message})
```

Comment:
Very minor comment, but this code (and registration) might be better living in webhooks.py.  (Similar to the way gcode.py provides both command handling code and a handful of low-level commands.)

Reply:
I'm okay with registering this in webhooks.py.  The only question I have is what should the API path be?  We probably don't want it under `webhooks/info`.  I could just register it as 'info', which would register it as `http://host/printer/info` on the server.

Reply:
I'm fine with `webhooks/info` or `info`.  We could also add a new startcmds.py and move many of the "commands needed before the config file" to that file (eg, cmd_m112(), restart, firmware_restart, CommandError, etc.).  Might be easier to merge first and then tackle that afterwards though.